api-tests/test_update_responses.py

```diff
@@ -0,0 +1,62 @@
+import requests
+from io import StringIO
+from lxml import etree
+
+
+def test_update_addon_responses():
+    addon_urls = [
+        'https://aus5.mozilla.org/update/3/SystemAddons/44.0/20160310153207/Darwin_x86_64-gcc3-u-i386-x86_64/en-GB/release/Darwin%2014.5.0/default/default/update.xml',
+    ]
+    dtd_info = """
+    <!ELEMENT updates (addons) >
+    <!ELEMENT addons (addon+) >
+    <!ELEMENT addon (#PCDATA)>
+    <!ATTLIST addon
+    id CDATA #REQUIRED
+    URL CDATA #REQUIRED
+    hashFunction CDATA #REQUIRED
+    hashValue CDATA #REQUIRED
+    size CDATA #REQUIRED
+    version CDATA #REQUIRED >
+    """
```

Comment:
It looks like this dtd doesn't work when no System Addons are returned, eg: https://aus5.mozilla.org/update/3/SystemAddons/38.0/20160310153207/Darwin_x86_64-gcc3-u-i386-x86_64/en-GB/release/Darwin%2014.5.0/default/default/update.xml, which is a valid response.